logic/atrium-logic-common/src/main/kotlin/ch/tutteli/atrium/logic/creating/basic/contains/creators/impl/ContainsAssertionCreator.kt

```diff
@@ -97,10 +97,9 @@ abstract class ContainsAssertionCreator<T : Any, TT : Any, in SC, C : Contains.C
                     assertionBuilder.explanatory
                         .withExplanation(descriptionNotFound)
                         .build()
-                ).let {
-                    if (checker.createAssertion(count).holds()) it
-                    else it.failing
-                }.build()
+                ).failing
+                .build()
+                .takeUnless { checker.createAssertion(count).holds() }
```

Comment:
to tie in with the bonus topic, this could look more or less like the following
```kotlin
           if (checker.createAssertion(count).holds()) {
                assertionBuilder.explanatoryGroup
                    .withDefaultType
                    .withAssertion(
                        assertionBuilder.explanatory
                            .withExplanation(descriptionNotFound)
                            .build()
                    )
                    .failing
                    .build()
            } else {
                //TODO replace by translatable with arguments
                assertionBuilder.explanatoryGroup
                    .withDefaultType
                    .withExplanatoryAssertion("and N such elements were found")
                    .build()
            }
```